server/example/workflow-modelserver-example/src/main/java/com/eclipsesource/glsp/example/modelserver/workflow/handler/AbstractCreateEdgeHandler.java

```diff
@@ -60,21 +61,37 @@ public String getLabel(AbstractOperationAction action) {
 	}
 
 	@Override
-	public void execute(AbstractOperationAction action, GraphicalModelState modelState) {
+	public void doExecute(AbstractOperationAction action, GraphicalModelState modelState,
+			WorkflowModelServerAccess modelAccess) throws Exception {
 		CreateConnectionOperationAction createConnectionAction = (CreateConnectionOperationAction) action;
-		WorkflowModelServerAccess modelAccess = ModelServerAwareModelState.getModelAccess(modelState);
 		WorkflowFacade workflowFacade = modelAccess.getWorkflowFacade();
 		Workflow workflow = workflowFacade.getCurrentWorkflow();
 
 		Flow flow = (Flow) CoffeeFactory.eINSTANCE.create(eClass);
 		flow.setSource(modelAccess.getNodeById(createConnectionAction.getSourceElementId()));
 		flow.setTarget(modelAccess.getNodeById(createConnectionAction.getTargetElementId()));
+
+		Command addCommand = AddCommand.create(modelAccess.getEditingDomain(), workflow,
+				CoffeePackage.Literals.WORKFLOW__FLOWS, flow);
+
+		createDiagramElement(workflowFacade, workflow, flow, createConnectionAction);
+
+		if (!modelAccess.edit(addCommand).thenApply(res -> res.body()).get()) {
+			throw new IllegalAccessError("Could not execute command: " + addCommand);
+		}
+	}
+
+	protected void createDiagramElement(WorkflowFacade facace, Workflow workflow, Flow flow,
+			CreateConnectionOperationAction createNodeOperationAction) {
 		workflow.getFlows().add(flow);
 
-		workflowFacade.findDiagram(workflow).ifPresent(diagram -> {
+		facace.findDiagram(workflow).ifPresent(diagram -> {
```

Comment:
Please create an issue for removing this workaround later, once we put the notation resource into the model server too.